cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -107,6 +110,7 @@
   private LeaderElection leaderElection;
   private ResourceCoordinator resourceCoordinator;
   private Cancellable coordinationSubscription;
+  private ExecutorService heartbeatsSubscriptionExecutor;
```

Comment:
Note that this only happens when using the ``DistributedStreamService`` along with the ``InMemoryNotificationService``. So we didn't see that issue in a cluster.